src/range_map.rs

```diff
@@ -84,11 +84,11 @@ impl<T> RangeMap<T> {
         slice
             .iter()
             .take_while(move |elem| elem.range.start < end)
-            .map(|elem| (Size::from_bytes(elem.range.start), &elem.data))
+            .map(|elem| (elem.range.clone(), &elem.data))
     }
 
-    pub fn iter_mut_all(&mut self) -> impl Iterator<Item = &mut T> {
-        self.v.iter_mut().map(|elem| &mut elem.data)
+    pub fn iter_mut_all(&mut self) -> impl Iterator<Item = (ops::Range<u64>, &mut T)> {
```

Comment:
Looks like no user of iter_mut_all actually needs the range, so maybe let's keep this how it used to be?

Reply:
More accurately,
- callers of `iter_mut_all` need nothing
- callers of `iter_all` need the offset
- callers of `iter_mut` need the range
- callers of `iter` need nothing

Should we go for the minimum signature for each, or something more consistent across the 4 iterators ?